### PR TITLE
tutanota-desktop: 3.119.3 -> 3.122.5, use appimage

### PR DESCRIPTION
**Pull request by WolfangAukang**
## Description of changes

https://github.com/tutao/tutanota/compare/tutanota-release-3.119.3...tutanota-release-3.122.5

Building with AppImage as it fixes Tutanota issues regarding unregistered modules (#221379, #278072).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (surfaceflinger) [inline pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:15]: Use just `extract`

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (surfaceflinger) [inline pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:45]: Check if Tutanota Desktop works fine with Wayland, if yes then keep makeWrapper and add `NIXOS_OZONE_WL` support

**Review by surfaceflinger — CHANGES_REQUESTED** [review]
You can avoid the whole let ... in like here https://github.com/NixOS/nixpkgs/pull/278137/files

Makes it more readable imo

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (WolfangAukang) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:45]: I do not have Wayland to test this

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (WolfangAukang) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:15]: Why use `extract` specifically? I ask this because `pkgs/by-name` has only two packages with `extract`:
```
grep -r 'appimageTools.extract' pkgs/by-name/
pkgs/by-name/hi/hifile/package.nix:  appimageContents = appimageTools.extractType2 {
pkgs/by-name/ca/caido/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname src version; };
# HERE
pkgs/by-name/lu/lunar-client/package.nix:  appimageContents = appimageTools.extract { inherit pname version src; };
pkgs/by-name/be/beekeeper-studio/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
pkgs/by-name/an/anytype/package.nix:  appimageContents = appimageTools.extractType2 { inherit name src; };
pkgs/by-name/ar/arduino-ide/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
pkgs/by-name/li/listen1/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
# HERE
pkgs/by-name/si/simplex-chat-desktop/package.nix:  appimageContents = appimageTools.extract {
pkgs/by-name/sp/spacedrive/package.nix:      appimageContents = appimageTools.extractType2 { inherit pname version src; };
```
Is there something on the documentation specifying this?

Comment (surfaceflinger) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix]: Documentation was updated recently, before that most maintainers just copied from each other outdated method. https://github.com/NixOS/nixpkgs/pull/276029

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (jtrees) [inline pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:8]: The most recent version is 3.120.6 by now btw.

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (jtrees) [inline pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:23]:
```suggestion
      install -Dm 444 ${appimageContents}/tutanota-desktop.png -t $out/share/pixmaps
```

**Review by jtrees — CHANGES_REQUESTED** [review]
Thanks you for this. The error box was really annoying.

Briefly tested on Hyprland (wayland). Seems to work fine. No more errors!

Just gotta fix the icon and you get green light from me.

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (jtrees) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:45]: The way it is now works for my on Hyprland with XWayland. Using the args provided by @CactiChameleon9 it does not start for me. I think the relevant log line is this:

```
[57048:0118/104949.744896:ERROR:gpu_process_host.cc(957)] GPU process exited unexpectedly: exit_code=139
```

Using mesa with an AMD card btw (don't know how relevant that is).

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (WolfangAukang) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix]: Got it, that makes sense. Updated.

File: pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix
Comment (jtrees) [reply pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix:45]: If you leave it in its current form (the one in this PR), it will continue to work for everyone, no? Save the wayland stuff for another PR.

**Comment by jtrees** [comment]
It seems Tutao is dropping support for 3.119.3 "within the next few days". Could we backport this to nixos-23.11?

**Pull request merged**
